cclient/serialization/json/get_transactions_to_approve.c

```diff
@@ -50,25 +45,77 @@ retcode_t json_get_transactions_to_approve_serialize_request(const serializer_t
   return ret;
 }
 
-retcode_t json_get_transactions_to_approve_deserialize_response(const serializer_t *const s, const char *const obj,
-                                                                get_transactions_to_approve_res_t *out) {
+retcode_t json_get_transactions_to_approve_deserialize_request(serializer_t const *const s, char const *const obj,
+                                                               get_transactions_to_approve_req_t *out) {
   retcode_t ret = RC_OK;
   cJSON *json_obj = cJSON_Parse(obj);
   cJSON *json_item = NULL;
 
   log_debug(json_logger_id, "[%s:%d] %s\n", __func__, __LINE__, obj);
   JSON_CHECK_ERROR(json_obj, json_item, json_logger_id);
 
-  ret = json_string_hash_to_flex_trits(json_obj, "trunkTransaction", out->trunk);
+  ret = json_get_uint32(json_obj, "depth", &out->depth);
   if (ret != RC_OK) {
     goto end;
   }
 
-  ret = json_string_hash_to_flex_trits(json_obj, "branchTransaction", out->branch);
+  ret = json_string_hash_to_flex_trits(json_obj, "reference", out->reference);
```

Comment:
`reference` is optional, please check if it is set.

Reply:
@HowJMay Please check with `cJSON_HasObjectItem` before